scene/main/viewport.cpp

```diff
@@ -682,32 +682,31 @@ void Viewport::_process_picking() {
 
 	PhysicsDirectSpaceState2D *ss2d = PhysicsServer2D::get_singleton()->space_get_direct_state(find_world_2d()->get_space());
 
-	if (physics_has_last_mousepos) {
+	bool has_mouse_event = false;
+	for (const Ref<InputEvent> &e : physics_picking_events) {
+		Ref<InputEventMouse> m = e;
+		if (m.is_valid()) {
+			has_mouse_event = true;
+			break;
+		}
+	}
+
+	if (!has_mouse_event) {
 		// If no mouse event exists, create a motion one. This is necessary because objects or camera may have moved.
 		// While this extra event is sent, it is checked if both camera and last object and last ID did not move.
 		// If nothing changed, the event is discarded to avoid flooding with unnecessary motion events every frame.
-		bool has_mouse_event = false;
-		for (const Ref<InputEvent> &m : physics_picking_events) {
-			if (m.is_valid()) {
-				has_mouse_event = true;
-				break;
-			}
-		}
+		Ref<InputEventMouseMotion> mm;
+		mm.instantiate();
 
-		if (!has_mouse_event) {
-			Ref<InputEventMouseMotion> mm;
-			mm.instantiate();
-
-			mm->set_device(InputEvent::DEVICE_ID_INTERNAL);
-			mm->set_global_position(physics_last_mousepos);
-			mm->set_position(physics_last_mousepos);
-			mm->set_alt_pressed(physics_last_mouse_state.alt);
-			mm->set_shift_pressed(physics_last_mouse_state.shift);
-			mm->set_ctrl_pressed(physics_last_mouse_state.control);
-			mm->set_meta_pressed(physics_last_mouse_state.meta);
-			mm->set_button_mask(physics_last_mouse_state.mouse_mask);
-			physics_picking_events.push_back(mm);
-		}
+		mm->set_device(InputEvent::DEVICE_ID_INTERNAL);
+		mm->set_position(get_mouse_position());
+		mm->set_global_position(mm->get_position());
+		mm->set_alt_pressed(Input::get_singleton()->is_physical_key_pressed(Key::ALT));
+		mm->set_shift_pressed(Input::get_singleton()->is_physical_key_pressed(Key::SHIFT));
+		mm->set_ctrl_pressed(Input::get_singleton()->is_physical_key_pressed(Key::CMD_OR_CTRL));
+		mm->set_meta_pressed(Input::get_singleton()->is_physical_key_pressed(Key::META));
```

Comment:
```suggestion
		mm->set_alt_pressed(Input::get_singleton()->is_key_pressed(Key::ALT));
		mm->set_shift_pressed(Input::get_singleton()->is_key_pressed(Key::SHIFT));
		mm->set_ctrl_pressed(Input::get_singleton()->is_key_pressed(Key::CTRL));
		mm->set_meta_pressed(Input::get_singleton()->is_key_pressed(Key::META));
```

I'm not super sure. What do you think?

Reply:
I'm not versed in these difference between `is_physical_key_pressed` and `is_key_pressed` and their implications, so I will accept your changes.
Have tested these changes with the MPRs of both linked issues.